### PR TITLE
 [query] Expose Table._map_partitions in Python

### PR DESCRIPTION
**Pull request by tpoterba**
stacked on #9141

### CODE REVIEW

**Comment by johnc1231** [comment]
Can be rebased / unstacked.

**Comment by tpoterba** [comment]
unstacked.

File: hail/python/hail/table.py [hunk 1]
```diff
@@ -3474,5 +3474,22 @@ def multi_way_zip_join(tables, data_field_name, global_field_name) -> 'Table':
...
+    @typecheck_method(f=func_spec(1, expr_array(expr_struct())))
+    def _map_partitions(self, f):
+        rows_uid = 'rows'
+        # rows_uid = Env.get_uid()
+        globals_uid = 'globals'
+        # globals_uid = Env.get_uid()
+        expr = construct_expr(ir.ToArray(ir.Ref(rows_uid)), hl.tarray(self.row.dtype), self._row_indices)
+        body = f(expr)
+        result_t = body.dtype
+        # if not isinstance(result_t, hl.tarray) and isinstance(result_t.element_type, hl.tstruct):
```
Comment (johnc1231) [inline hail/python/hail/table.py:3486]: Remove or uncomment?

Comment (tpoterba) [reply hail/python/hail/table.py:3486]: oh, remove. I added the typechecker! :D

**Comment by johnc1231** [comment]
Thanks for adding this!

**Pull request merged**